### DISCUSSION TITLE
is `engine(input)` the same as running each part of the model separately?

### DISCUSSION THREAD

**Discussion by tnq177** [deepspeedai/DeepSpeed#2378] [Q&A]
I apologize if this question is too dumb. I'm trying to use DeepSpeed with a pretrained XLM-R model. Upon inspecting, the model consists of 2 modules: `robert` (the transformer layers) and `lm_head` (the softmax layer). Is running:

```
output = engine(input)
```
the same as:
```
hidden = engine.roberta(input)
output = engine.lm_head(hidden)
```

Thanks!
